contrib/kubespray/script/openpai-generator.py

```diff
@@ -153,13 +154,51 @@ def get_kubernetes_node_info_from_API():
     return ret
 
 
+def get_pai_daemon_resource_request(cfg):
+    ret = {
+        "cpu-resource": 0,
+        "mem-resource": 0
+    }
+    if "qos-switch" not in cfg or cfg["qos-switch"] == "false":
+        logger.info("Ignore calculate pai daemon resource usage since qos-switch set to false")
+        return ret
+
+    pai_daemon_services = ["node-exporter", "job-exporter", "log-manager"]
+    pai_source_path = os.path.join(os.path.dirname(os.path.realpath(__file__)), "../../../src")
+
+    # {%- if cluster_cfg['cluster']['common']['qos-switch'] == "true" %}
+    start_match = r"{%-?\s*if\s*cluster_cfg\['cluster'\]\['common'\]\['qos-switch'\][^}]+%}"
+    end_match = r"{%-?\s*endif\s*%}"  # {%- end %}
+    str_match = "{}(.*?){}".format(start_match, end_match)
+    regex = re.compile(str_match, flags=re.DOTALL)
+
+    for pai_daemon in pai_daemon_services:
+        deploy_template_path = os.path.join(pai_source_path, "{0}/deploy/{0}.yaml.template".format(pai_daemon))
+        if os.path.exists(deploy_template_path):
+            template = read_template(deploy_template_path)
+            match = regex.search(template)
+            if not match:
+                logger.warning("Could not find resource request for service %s", pai_daemon)
+                continue
+            resources = yaml.load(match.group(1), yaml.SafeLoader)["resources"]
+            if "requests" in resources:
+                ret["cpu-resource"] += parse_quantity(resources["requests"].get("cpu", 0))
+                ret["mem-resource"] += parse_quantity(resources["requests"].get("memory", 0)) / 1024 / 1024
```

Comment:
Can we use math.ceil to an int?

Reply:
When we generate hived config, we will change it to int.
Refer: https://github.com/microsoft/pai/blob/364aee769a421771ab76284d55fd7039767d3f20/contrib/kubespray/script/openpai-generator.py#L200-L201

Reply:
ok